solidity/test/bridge/Bridge.Governance.test.ts

```diff
@@ -0,0 +1,4318 @@
+        ).to.be.revertedWith("Ownable: caller is not the owner")
+      })
+    })
+
+    context("when the update process is not initialized", () => {
+      it("should revert", async () => {
+        await expect(
+          bridgeGovernance
+            .connect(governance)
+            .finalizeFraudNotifierRewardMultiplierUpdate()
+        ).to.be.revertedWith("Change not initiated")
+      })
+    })
+
+    context("when the governance delay has not passed", () => {
+      before(async () => {
+        await createSnapshot()
+
+        await bridgeGovernance
+          .connect(governance)
+          .beginFraudNotifierRewardMultiplierUpdate(42)
+
+        await helpers.time.increaseTime(constants.governanceDelay - 60) // -1min
+      })
+
+      after(async () => {
+        await restoreSnapshot()
+      })
+
+      it("should revert", async () => {
+        await expect(
+          bridgeGovernance
+            .connect(governance)
+            .finalizeFraudNotifierRewardMultiplierUpdate()
+        ).to.be.revertedWith("Governance delay has not elapsed")
+      })
+    })
+
+    context(
+      "when the update process is initialized and governance delay passed",
+      () => {
+        let tx: ContractTransaction
+
+        before(async () => {
+          await createSnapshot()
+
+          await bridgeGovernance
+            .connect(governance)
+            .beginFraudNotifierRewardMultiplierUpdate(42)
+
+          await helpers.time.increaseTime(constants.governanceDelay)
+
+          tx = await bridgeGovernance
+            .connect(governance)
+            .finalizeFraudNotifierRewardMultiplierUpdate()
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        it("should update the fraud notifier reward multiplier", async () => {
+          const { fraudNotifierRewardMultiplier } =
+            await bridge.fraudParameters()
+          expect(fraudNotifierRewardMultiplier).to.be.equal(42)
+        })
+
+        it("should emit FraudNotifierRewardMultiplierUpdated event", async () => {
+          await expect(tx)
+            .to.emit(bridgeGovernance, "FraudNotifierRewardMultiplierUpdated")
+            .withArgs(42)
+        })
+      }
+    )
+  })
+
+  describe("setVaultStatus", () => {
+    context("when the caller is not the owner", () => {
+      it("should revert", async () => {
+        await expect(
+          bridgeGovernance
+            .connect(thirdParty)
+            .setVaultStatus(thirdParty.address, true)
+        ).to.be.revertedWith("Ownable: caller is not the owner")
+      })
+    })
+
+    context("when the caller is the owner", () => {
+      it("should emit VaultStatusUpdated event", async () => {
+        const tx = await bridgeGovernance
+          .connect(governance)
+          .setVaultStatus(thirdParty.address, true)
+
+        await expect(tx)
+          .to.emit(bridge, "VaultStatusUpdated")
+          .withArgs(thirdParty.address, true)
+      })
```

Comment:
We could check that the statue was actually changed in the Bridge. So we should change the context to something like:

```
    context("when the caller is the owner", () => {
      let tx: ContractTransaction

      before(async () => {
        await createSnapshot()

        tx = await bridgeGovernance
          .connect(governance)
          .setVaultStatus(thirdParty.address, true)
      })

      after(async () => {
        await restoreSnapshot()
      })

      it("should mark the vault as trusted", async () => {
        expect(await bridge.isVaultTrusted(thirdParty.address)).to.be.true
      })

      it("should emit VaultStatusUpdated event", async () => {
        await expect(tx)
          .to.emit(bridge, "VaultStatusUpdated")
          .withArgs(thirdParty.address, true)
      })
    })
```

Reply:
[233fb72](https://github.com/keep-network/tbtc-v2/pull/329/commits/233fb72896adee24468a24883f6073bcab073865)